### PR TITLE
Add/timereftype

### LINKED ISSUE #60: Extract complex VectorData to its own type: TimeSeriesRef
The `IntracellularStimuliTable` and `IntracellularResponsesTable` types both have a dataset:
```yaml
- name: stimulus
  neurodata_type_inc: VectorData
  dtype:
  - name: idx_start
    dtype: int32
    doc: Start index into the TimeSeries 'data' and 'timestamp' datasets of the
      referenced TimeSeries. The first dimension of those arrays is always time.
  - name: count
    dtype: int32
    doc: Number of data samples available in this time series, during this epoch
  - name: timeseries
    dtype:
      target_type: TimeSeries
      reftype: object
    doc: The TimeSeries that this index applies to
  doc: Column storing the reference to the recorded stimulus for the recording
    (rows)
```

Aside from the name and docstring, this is the same as the definition for the `'timeseries'` column of the `TimeIntervals` table. Since this exact structure would now be used three times in the schema, I think this should be its own type named something like `TimeSeriesRef` or `TimeSeriesColumn` and included in the table, e.g.:
```yaml
- name: stimulus
  neurodata_type_inc: TimeSeriesRef
  doc: Column storing the reference to the recorded stimulus for the recording
    (rows)
```

### PR DESCRIPTION
**Pull request by oruebel**
Fix #60 Add TimeSeriesReferenceVectorData type

### CODE REVIEW

**Comment by rly** [comment]
https://github.com/hdmf-dev/hdmf/pull/325 needs to be merged to HDMF first

**Comment by oruebel** [comment]
hdmf-dev/hdmf#325  has been closed because the feature has already been added to HDMF dev, but instead of ``colcls``, the argument is named ``col_cls``. This PR needs to be updated accordingly.

**Comment by rly** [comment]
This PR works now. Running the tests results in four errors which are also present in the master branch (see https://github.com/oruebel/ndx-icephys-meta/issues/77).

I am not sure if my fixes to `test_icephys.ICEphysFileTests.test_add_icephys_meta_full_roundtrip` are appropriate. Please verify @oruebel.

File: src/pynwb/ndx_icephys_meta/test/test_icephys.py [hunk 1]
```diff
@@ -1243,7 +1244,7 @@ def test_add_icephys_meta_full_roundtrip(self):
         # check the id value
         self.assertEqual(res.index[0], 12)
         # Check that our sweep contains 1 IntracellularRecording
-        self.assertEqual(len(res.iloc[0]['recordings']), 1)
+        self.assertEqual(len(res), 1)
```
Comment (oruebel) [inline src/pynwb/ndx_icephys_meta/test/test_icephys.py:1247]: @rly, while this is valid assert to make, it changes the meaning of what is being tested. The assert `` self.assertEqual(len(res), 1)`` checks that the table has a whole only a single row, whereas ``self.assertEqual(len(res.iloc[0]['recordings']), 1)`` tries to check whether the first cell in the column ``'recordings'`` contains a single element. The ``'recordings'`` column in this case is ragged array column and so the assert was trying to check whether the recording exists and that we have only 1. The same is true in line 1264, 1280, 1296, 1392, 1403, 1415, and 1426.

**Comment by oruebel** [comment]
@rly aside from the comment regarding the test, this looks good to me.

**Comment by rly** [comment]
@oruebel Thanks. I fixed those tests and to test the raggedness, I changed the first test so that one simultaneous recording includes more than one intracellular recording. I get the error:
```
Traceback (most recent call last):
  File "C:\Users\Ryan\Documents\NWB\ndx-icephys-meta\src\pynwb\ndx_icephys_meta\test\test_icephys.py", line 1247, in test_add_icephys_meta_full_roundtrip
    res = nwbfile.icephys_simultaneous_recordings[0]
  File "c:\users\ryan\documents\nwb\hdmf\src\hdmf\common\table.py", line 723, in __getitem__
    ret = self.get(key)
  File "c:\users\ryan\documents\nwb\hdmf\src\hdmf\common\table.py", line 829, in get
    ret = pd.DataFrame(retdf, index=pd.Index(name=self.id.name, data=id_index))
  File "C:\Users\Ryan\miniconda3\envs\dev\lib\site-packages\pandas\core\frame.py", line 411, in __init__
    mgr = init_dict(data, index, columns, dtype=dtype)
  File "C:\Users\Ryan\miniconda3\envs\dev\lib\site-packages\pandas\core\internals\construction.py", line 257, in init_dict
    return arrays_to_mgr(arrays, data_names, index, columns, dtype=dtype)
  File "C:\Users\Ryan\miniconda3\envs\dev\lib\site-packages\pandas\core\internals\construction.py", line 87, in arrays_to_mgr
    return create_block_manager_from_arrays(arrays, arr_names, axes)
  File "C:\Users\Ryan\miniconda3\envs\dev\lib\site-packages\pandas\core\internals\managers.py", line 1699, in create_block_manager_from_arrays
    construction_error(len(arrays), arrays[0].shape, axes, e)
  File "C:\Users\Ryan\miniconda3\envs\dev\lib\site-packages\pandas\core\internals\managers.py", line 1719, in construction_error
    "Shape of passed values is {0}, indices imply {1}".format(passed, implied)
ValueError: Shape of passed values is (2, 1), indices imply (1, 1)
```

I thought that these tables should create multi-indexes by default. Does it not? 

The functionality of `DynamicTable.get` had been modified several times over the past year, so perhaps this line used to work and some behavior needs to be reverted or changed.

**Comment by oruebel** [comment]
> I thought that these tables should create multi-indexes by default. Does it not?

Yes, normally this should automatically create the ragged column since this is prescribed that way in the schema. 

> so perhaps this line used to work and some behavior of `DynamicTable.get` needs to be reverted or changed.

I'm pretty sure this worked at some point, but it certainly possible that it needs to be updated to deal with updated behavior in HDMF.

**Pull request merged**